lua_modules/liquidcrystal/i2c4bit.lua

```diff
@@ -0,0 +1,59 @@
+local I2C4bit = {}
+
+local function send4bitI2C(value, cmd, bus_args, backlight)
+   local hi = bit.bor(bit.lshift(bit.band(bit.rshift(value, 4), 1), bus_args.pinout.d4),
+		      bit.lshift(bit.band(bit.rshift(value, 5), 1), bus_args.pinout.d5),
+		      bit.lshift(bit.band(bit.rshift(value, 6), 1), bus_args.pinout.d6),
+		      bit.lshift(bit.band(bit.rshift(value, 7), 1), bus_args.pinout.d7))
+   local lo = bit.bor(bit.lshift(bit.band(bit.rshift(value, 0), 1), bus_args.pinout.d4),
+		      bit.lshift(bit.band(bit.rshift(value, 1), 1), bus_args.pinout.d5),
+		      bit.lshift(bit.band(bit.rshift(value, 2), 1), bus_args.pinout.d6),
+		      bit.lshift(bit.band(bit.rshift(value, 3), 1), bus_args.pinout.d7))
+   if backlight then
+      cmd = bit.bor(cmd, bit.lshift(1, bus_args.pinout.backlight))
+   end
+   i2c.start(bus_args.id)
+   i2c.address(bus_args.id, bus_args.address, i2c.TRANSMITTER)
+   i2c.write(bus_args.id, bit.bor(hi, cmd, bit.lshift(1, bus_args.pinout.en)))
+   i2c.write(bus_args.id, bit.bor(hi, bit.band(cmd, bit.bnot(bit.lshift(1, bus_args.pinout.en)))))
+   i2c.write(bus_args.id, bit.bor(lo, cmd, bit.lshift(1, bus_args.pinout.en)))
+   i2c.write(bus_args.id, bit.bor(lo, bit.band(cmd, bit.bnot(bit.lshift(1, bus_args.pinout.en)))))
+   i2c.stop(bus_args.id)
+end
+
+I2C4bit.command = function(self, value)
+   send4bitI2C(value, 0x0, self._bus_args, self._backlight)
+end
+
+I2C4bit.write = function(self, value)
+   send4bitI2C(value, bit.lshift(1, self._bus_args.pinout.rs), self._bus_args, self._backlight)
+end
+
+I2C4bit.backlight = function(self, on)
+   self._backlight = on
+   self._backend.command(self, 0x0)
+end
+
+I2C4bit.init = function(self)
+   if self._bus_args.pinout == nil then
+      self._bus_args.pinout = {
```

Comment:
Thanks, but I think making this a second table is bloating memory, which you indicated was tight.  Maybe something like
```
  self._bus_args.rs = self._bus_args.rs or 0
  self._bus_args.rw = self._bus_args.rw or 1
  -- ...
```